service/src/database/dips.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023-, Edge & Node, GraphOps, and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+use axum::async_trait;
+
+use crate::routes::dips::Agreement;
+
+#[async_trait]
+pub trait AgreementStore: Sync + Send {
+    async fn get_by_signature(&self, signature: String) -> anyhow::Result<Option<Agreement>>;
```

Comment:
I am considering using a UUIDv7 as the agreement identifier instead of the signature. It has some good properties that we can use to manage and trace data across the different DIPs' parties (i.e., DIPs Gateway/Dipper, Indexer, and future smart contracts).

Of course, these IDs will be only relevant on the indexer-facing side of the DIPs domain.

WDYT?

Reply:
For example, as UUIDv7 is lexicographically sortable, we can quickly sort the agreements from oldest to newest without requiring a timestamp field.